### ISSUE TITLE
More flexible interpretation of mutation rate when getting emission probability when using `_tskit.lshmm`

### ISSUE THREAD

**Issue by szhan** [tskit-dev/tskit#2891]
Currently, the emission probability for calculating the forward, backward, and Viterbi matrices is defined as follows:

```
p_e = mu;
if (is_match) {
    p_e = 1 - (num_alleles - 1) * mu;
}
```

This implies that an allele at a site mutates to another allowed allele (`num_alleles - 1`) uniformly at random at a user-specified mutation rate, mu. Changing the formula for `p_e` to `p_e = 1 - mu` upon allele match should allow for more flexible (general?) interpretation of the mutation rate.

Relevant lines are the following:
https://github.com/tskit-dev/tskit/blob/2dae133fd8c3414d2f9b5073b964e37b4f96e032/c/tskit/haplotype_matching.c#L1055
https://github.com/tskit-dev/tskit/blob/2dae133fd8c3414d2f9b5073b964e37b4f96e032/c/tskit/haplotype_matching.c#L1106
https://github.com/tskit-dev/tskit/blob/2dae133fd8c3414d2f9b5073b964e37b4f96e032/c/tskit/haplotype_matching.c#L1295

**Comment by szhan** [comment]
I should also note that in the BEAGLE implementation of the LS HMM, it is `p_e = 1 - mu` upon allele match.

**Comment by szhan** [comment]
Shouldn't the probability of match and probability of mismatch sum to 1? If that is the case, then I suspect there is a major bug.

In this code snippet:
```
p_e = mu;
if (is_match) {
    p_e = 1 - (num_alleles - 1) * mu;
}
```

Probability of mismatch = `mu`
Probability of match = `1 - (num_alleles - 1) * mu`

Their sum is clearly not 1.

Probability of match should be `(1 - mu)`, unless probability of mismatch is `(num_alleles - 1) * mu`.

**Comment by szhan** [comment]
I'd also like to add a note that BEAGLE 4.1 defines probability of mismatch and probability of match as `mu` and `1 - mu`, respectively. The BEAGLE 4.1 paper analyses only biallelic sites.

**Comment by jeromekelleher** [comment]
I agree it is a weird parameterisation, and doesn't belong in the core HMM. We should make this type of rescaling optional at the Python level.

**Comment by szhan** [comment]
I'm just looking at how computation of emission probability is done in `test_haplotype_matching.py`. It seems a bit different. Is it going with a slightly different interpretation of `mu`, when there is no scaling based on the number of alleles?

```
if is_match:
    p_e = 1 - mu
else:
    p_e = mu / (n_alleles - 1)
```

https://github.com/tskit-dev/tskit/blob/2dae133fd8c3414d2f9b5073b964e37b4f96e032/python/tests/test_haplotype_matching.py#L382

**Comment by jeromekelleher** [comment]
Maybe ``mu`` has been rescaled somewhere else? There may well be inconsistencies here, though

**Comment by benjeffery** [comment]
Closing for inactivity and labelling "future", please re-open if you plan to work on this.
